lib/tty/multi_spinner.rb

```diff
@@ -0,0 +1,33 @@
+require 'tty/spinner'
+
+module TTY
+  # Used for managing multiple terminal spinners
+  #
+  # @api public
+  class MultiSpinner
+    def initialize(options = {})
+      @options = options
+
+      @spinners = []
+    end
+
+    def add_spinner(message, options = {})
+      new_spinner = TTY::Spinner.new(message, @options.merge(options))
+      new_spinner.add_multispinner(self, @spinners.length)
+      @spinners.push new_spinner
+
+      new_spinner
+    end
+
+    def lines_between_end_and_index(index)
+      count = 0
+      @spinners.each_with_index do |spinner, i|
+        next if i < index
+
+        count += 1 if spinner.spinning? || spinner.success? || spinner.error? || spinner.done?
```

Comment:
Correct, spinning and done are straightforward, and when the spinner is entering a "success" or "error" state it is not yet "done" so those two are needed so the success/error message is printed on the correct line.